packages/adapters/bitcoin/src/adapter.ts

```diff
@@ -169,11 +178,30 @@ export class BitcoinAdapter extends AdapterBlueprint<BitcoinConnector> {
 
   // -- Unused => Refactor ------------------------------------------- //
 
-  override getBalance(
-    _params: AdapterBlueprint.GetBalanceParams
+  override async getBalance(
+    params: AdapterBlueprint.GetBalanceParams
   ): Promise<AdapterBlueprint.GetBalanceResult> {
+    const network = params.caipNetwork
+
+    if (network?.chainNamespace === 'bip122') {
```

Comment:
why do we need this check?

Reply:
it validates type safety of `caipNetwork` that may be undefined and plus checks if is `bip122` to guarantee it not breaking